app/src/main/java/org/schabi/newpipe/fragments/MainFragment.java

```diff
@@ -18,11 +18,16 @@
 
 import org.schabi.newpipe.BaseFragment;
 import org.schabi.newpipe.R;
+import org.schabi.newpipe.fragments.list.kisok.KioskFragment;
 import org.schabi.newpipe.fragments.subscription.SubscriptionFragment;
 import org.schabi.newpipe.util.NavigationHelper;
 
 public class MainFragment extends BaseFragment implements TabLayout.OnTabSelectedListener {
     private ViewPager viewPager;
+    private boolean showBlankTab = false;
+
+    //todo: FIX THIS URGENTLY
+    public int currentServiceId = 0; //for youtube
```

Comment:
If #693 is merged, you could do `Integer.parseInt(PreferenceManager.getDefaultSharedPreferences(getActivity()).getString("service", "0"))`.

Reply:
Yeah, but #693 isn't merged yet, so you'd have to wait for that.

Reply:
hm, don't you think it was possible to introduc it anyway, I mean its no code that will colide i guess.

Reply:
You mean merging #693? Well, that's possible, but then I'd at least add the correct intents for SoundCloud (can you do that? I know nothing about Android's intents xD). After that I can just make another PR for the new service layout.

Reply:
No don't merge #693 sorry. No I ment adding that line `Integer.parseInt(PreferenceManager.getDefaultSharedPreferences(getActivity()).getString("service", "0")).` wouldn't make any code collide. So we could put it into fragment_main anyway. ... What do you think?

Reply:
Oh, if that doesn't throw an error, you should add it.

Reply:
It don't, but we are hardcoding keys (see settings_key.xml). Also, why are we parsing a `String` when we could save the `int` directly?

Reply:
@mauriciocolli: I tried doing it with `int`s, but I got some weird error. If you could get it working, please commit that to #693.

Reply:
Well we we could put it into settings_keys.xml than its just one line that collides.